src/client/connection.ts

```diff
@@ -43,40 +44,56 @@ export function getEventRouter(): EventRouter<Events> {
 /**
  * Promise to the channel object that allows us to connect to the client
  */
-let channelPromise: Promise<ChannelClient>|null = null;
+let channelPromise: Promise<ChannelClient> | null = null;
+const hasDOMContentLoaded = new DeferredPromise<void>();
 
-if (typeof fin !== 'undefined') {
-    getServicePromise();
-}
+if (typeof document !== 'undefined') {
```

Comment:
When would `document` be undefined, and should we set a rjected value for `channelPromise` in that case?